syntax = "proto3";

package endpoints;

// This config is included for potential future use.
// In the future it may include part count, expiration, or other metadata.
message CreateArtifactConfig {
}

message BatchCreateArtifactRequest {
  // We are using a map here to avoid depending on the result array
  // being in the same order as the request array.
  map<int32, CreateArtifactConfig> configs_by_index = 1;
}

message CreateArtifactResult {
  optional string artifact_id = 1;
}

message BatchCreateArtifactResponse {
  map<int32, CreateArtifactResult> results_by_index = 1;
}
